prime-router/src/main/kotlin/azure/BatchDeciderFunction.kt

```diff
@@ -42,8 +43,17 @@ class BatchDeciderFunction : Logging {
                         // TODO: to support sending empty batches, we will need to check if the receiver wants that
                         //  and add the queue message regardless. Currently functionality only adds a message if
                         //  there is at least one batch to run
+
+                        // Calculate how far to look back based on how often this receiver batches.
```

Comment:
This is the main change - both here and in BatchFunction, calculate the backstop time on a per-receiver basis, rather than a global constant.   This lets us wait days for batches that only run once a day, and minutes for batches that run every minute, and minimizes the "death spiral" problem without having to invent yet another retry mechanism.